flake8_trio/visitors/visitor91x.py

```diff
@@ -494,10 +589,31 @@ def leave_While_orelse(self, node: cst.While | cst.For):
 
         # reset break & continue in case of nested loops
         self.outer[node]["uncheckpointed_statements"] = self.uncheckpointed_statements
-        self.restore_state(node)
```

Comment:
needed to be moved to `leave_While` since it has to be done after inserting checkpoints in the loop body.